adapters/syslog/syslog.go

```diff
@@ -96,13 +97,90 @@ func (a *SyslogAdapter) Stream(logstream chan *router.Message) {
 		_, err = a.conn.Write(buf)
 		if err != nil {
 			log.Println("syslog:", err)
-			if reflect.TypeOf(a.conn).String() != "*net.UDPConn" {
-				return
+			switch a.conn.(type) {
+			case *net.UDPConn:
+				continue
+			default:
+				err = a.retry(buf, err)
+				if err != nil {
+					log.Println("syslog:", err)
+					return
+				}
 			}
 		}
 	}
 }
 
+func (a *SyslogAdapter) retry(buf []byte, err error) error {
```

Comment:
You may as well fix the connection leak that exists her as well and do an a.conn.Close() if we decide that a connection is dead and go to reconnect.